test/data/austin.json

```diff
@@ -1 +1 @@
\ No newline at end of file
+{"$schema": "https://www.speedscope.app/file-format-schema.json", "shared": {"frames": [{"name": "_bootstrap", "file": "/usr/lib/python3.8/threading.py", "line": 890}, {"name": "_bootstrap_inner", "file": "/usr/lib/python3.8/threading.py", "line": 932}, {"name": "run", "file": "/usr/lib/python3.8/threading.py", "line": 870}, {"name": "keep_cpu_busy", "file": "/home/gabriele/Projects/austin/test/target34.py", "line": 32}, {"name": "<module>", "file": "/home/gabriele/Projects/austin/test/target34.py", "line": 37}, {"name": "start", "file": "/usr/lib/python3.8/threading.py", "line": 857}, {"name": "wait", "file": "/usr/lib/python3.8/threading.py", "line": 558}, {"name": "wait", "file": "/usr/lib/python3.8/threading.py", "line": 302}, {"name": "<module>", "file": "/home/gabriele/Projects/austin/test/target34.py", "line": 38}, {"name": "keep_cpu_busy", "file": "/home/gabriele/Projects/austin/test/target34.py", "line": 31}, {"name": "keep_cpu_busy", "file": "/home/gabriele/Projects/austin/test/target34.py", "line": 30}]}, "profiles": [{"name": "Wall time profile for 82848:82848", "unit": "microseconds", "startValue": 0, "endValue": 372856, "samples": [[4, 5, 6, 7], [8, 3], [8, 9], [8, 3], [8, 9], [8, 3], [8, 10], [8, 3], [8, 3], [8, 9], [8, 10], [8, 9], [8, 3], [8, 9], [8, 3], [8, 10], [8, 3], [8, 9], [8, 3], [8, 3], [8, 3], [8, 3], [8, 9], [8, 3], [8, 9], [8, 3], [8, 9], [8, 3], [8, 9], [8, 9], [8, 9], [8, 9], [8, 3], [8, 3], [8, 3], [8, 3], [8]], "weights": [10058, 10083, 10088, 10088, 10087, 10060, 10085, 10082, 10020, 10090, 10057, 10058, 10087, 10087, 10083, 10060, 10057, 10083, 10094, 10058, 10058, 10057, 10081, 10087, 10087, 10087, 10081, 10088, 10084, 10089, 10089, 10084, 10088, 10090, 10091, 10058, 10092], "type": "sampled"}, {"name": "Wall time profile for 82848:82858", "unit": "microseconds", "startValue": 0, "endValue": 352706, "samples": [[0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 9], [0, 1, 2, 9], [0, 1, 2, 9], [0, 1, 2, 9], [0, 1, 2, 3], [0, 1, 2, 9], [0, 1, 2, 3], [0, 1, 2, 9], [0, 1, 2, 9], [0, 1, 2, 9], [0, 1, 2, 9], [0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 9], [0, 1, 2, 10], [0, 1, 2, 9], [0, 1, 2, 9], [0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 9], [0, 1, 2, 9], [0, 1, 2, 3], [0, 1, 2, 9], [0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 9], [0, 1, 2, 10], [0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2, 3], [0, 1, 2], [0, 1, 2]], "weights": [10058, 10083, 10088, 10088, 10087, 10060, 10085, 10082, 10020, 10090, 10057, 10058, 10087, 10087, 10083, 10060, 10057, 10083, 10094, 10058, 10058, 10057, 10081, 10087, 10087, 10087, 10081, 10088, 10084, 10089, 10089, 10084, 10088, 10090, 10091], "type": "sampled"}], "name": "austin.out", "exporter": "Austin2Speedscope Converter 0.2.0"}
```

Comment:
This file was originally generated by calling `austin2speedscope` on `test/data/austin.out` with

~~~ console
poetry run austin2speedscope test/data/austin.out test/data/austin.json
~~~

 and was not updated when the latter was changed. I would have expected simply regenerating this file and making sure that it looks reasonable on Speedscope.app would have done the trick. This is what the pprof test does anyway, given the complexity of these file formats, even for relatively simple examples.

Aside: there should probably be a task in the `noxfile` to generate these snapshots when `austin.out` is changed. I'll look into this.

Reply:
As far as I understand it, I did something equivalent to regenerate `austin.json` from `austin.out` in [`6b1554cc`](https://github.com/P403n1x87/austin-python/pull/7/commits/6b1554cccbaf25b438378d063b02292f9ff4b3ac).:

```bash
poetry shell
austin2speedscope test/data/austin.out test/data/austin.json
```

Reply:
If there's anything else I should do re: the `austin.json`, let me know and I'll be happy to make the necessary changes.

Reply:
Ah, my bad! I've missed that the first time around! I now realise that `test_speedscope_wall_metrics_only` tests against the updated snapshot, so as long as the new `austin.json` looks reasonable we should be good! 👍 

Just as a nit, we could regenerate `austin.json` with e.g. `indent=2` so that it is more readable and easier to use to spot potential regressions in the future. But I leave this up to you.

Reply:
I agree with your suggestion re: indentation.

In [`9a56b7f8`](https://github.com/P403n1x87/austin-python/pull/7/commits/9a56b7f8af1f9ad1e088c02238ae000da74df46b) I've updated the `test_speedscope_wall_metrics_only` test to use `json.load` and `json.dumps`, which makes the string comparison in the assertion insensitive to whitespace in `austin.json`. This change made it easier to change the indentation in `austin.json` without breaking existing tests.

I then made the change to add indentation to `austin.json` -- as you suggested -- in [`43e70bd5`](https://github.com/P403n1x87/austin-python/pull/7/commits/43e70bd5bf80f9744e7ec701d2e11030795b42f5).